server/api/search.py

```diff
@@ -58,3 +60,45 @@ def search_cuisines():
     # All good, make the response and return
     response = jsonify(response_dict)
     return response, 200
+
+@search_views.route("/location", methods=["GET"])
+@authenticate
+def chefs_by_location(**kwargs):
+    """
+    Use the user object from kwargs to get the center point
+    of the radius then use sqlalchemy/haversine formula
+    to find all chefs within the given radius
+    """
+    response_dict = {}
+    user = kwargs['user']
+    request_dict = request.get_json()
+    distance = request_dict['distance']
+    chefs = session.query(User).join(User.profile).filter(
+        (func.degrees(
+            func.acos(
+                func.sin(func.radians(user.latitude)) * func.sin(func.radians(User.latitude))
+                + func.cos(func.radians(user.latitude)) * func.cos(func.radians(User.latitude))
+                * func.cos(func.radians(user.longitude-User.longitude))
+            )
+        ) * 60 * 1.1515 * 1.609344) <= distance).filter(Profile.is_chef == True).filter(User.id!=user.id).all()
```

Comment:
I confess I'm not super familiar with the underlying logic of the math, but this is different from the method I'm used to for calculating distance from latitude and longitude. Have you confirmed this is returning accurate results?

Reply:
yes, I tested it while in the account of the user with the email "foodgood@aol.com" (which is located in Minnesota) and it returned the only chef in the state. I can add a new chef in some other state and test it again though. Just realized there's only one chef so I'll retest this and let you know